widget/embedded/src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,20 @@
+import type { Notification } from '../types/notification';
+
+import { useNavigate } from 'react-router-dom';
+
+import { navigationRoutes } from '../constants/navigationRoutes';
+import { useNotificationStore } from '../store/notification';
+import { useUiStore } from '../store/ui';
+
+export function useNotifications() {
+  const setSelectedSwap = useUiStore.use.setSelectedSwap();
+  const { getUnreadNotifications } = useNotificationStore();
+  const navigate = useNavigate();
+  const notifications: Notification[] = getUnreadNotifications();
+  const onSelect = (requestId: Notification['requestId']) => {
+    setSelectedSwap(requestId);
```

Comment:
Considering that we are providing access to this function for external dapps, it would be prudent to include a validation check for the 'requestId'.

Reply:
fixed.